src/lib.rs

```diff
@@ -1 +1,199 @@
+// Disable warning for the crate name, not a really good way to do this but..
+// (https://github.com/rust-lang/rust/issues/45127)
 #![allow(non_snake_case)]
+
+use std::ffi::{c_void, OsStr, OsString};
+use std::mem::size_of;
+use std::path::{Path, PathBuf};
+use thiserror::Error;
+use windows::Win32::Foundation::{CloseHandle, GetLastError, PWSTR, STATUS_PENDING};
+use windows::Win32::Security::SECURITY_ATTRIBUTES;
+use windows::Win32::System::Threading::{
+    GetExitCodeProcess, TerminateProcess, WaitForSingleObject, PROCESS_CREATION_FLAGS,
+    PROCESS_INFORMATION, STARTUPINFOW, WAIT_OBJECT_0,
+};
+use windows::Win32::System::WindowsProgramming::INFINITE;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("cannot create process: {0}")]
+    CreationFailed(u32),
+    #[error("cannot get exit status: {0}")]
+    GetExitCodeFailed(u32),
+    #[error("cannot kill process: {0}")]
+    KillFailed(u32),
+}
+
+type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug)]
+pub struct Command {
+    command: OsString,
+    inherit_handles: bool,
+    current_directory: Option<PathBuf>,
+}
+
+impl Command {
+    pub fn new(command: impl Into<OsString>) -> Self {
+        Self {
+            command: command.into(),
+            inherit_handles: true,
+            current_directory: None,
+        }
+    }
+
+    pub fn inherit_handles(&mut self, inherit: bool) -> &mut Self {
+        self.inherit_handles = inherit;
+        self
+    }
+
+    pub fn current_dir(&mut self, dir: impl Into<PathBuf>) -> &mut Self {
+        self.current_directory = Some(dir.into());
+        self
+    }
+
+    pub fn spawn(&mut self) -> Result<Child> {
+        Child::new(
+            &self.command,
+            self.inherit_handles,
+            self.current_directory.as_deref(),
+        )
+    }
+
+    pub fn status(&mut self) -> Result<ExitStatus> {
+        self.spawn()?.wait()
+    }
+}
+
+#[derive(Debug)]
+pub struct Child {
+    process_information: PROCESS_INFORMATION,
+}
+
+impl Child {
+    fn new(
+        command: &OsStr,
+        inherit_handles: bool,
+        current_directory: Option<&Path>,
+    ) -> Result<Self> {
+        let mut startup_info = STARTUPINFOW::default();
```

Comment:
I would start unsafe here in the beginning as the entire logic is one block here